ls1intum/Artemis
# `General`: Only show learning goals when data is available

### Checklist
#### General


#### Server
#### Client

### Motivation and Context



The learning goals tab is shown, even when the instructor had not defined any learning goals or prerequisites for the course.

### Description


This PR fixes this issue by only showing the learning goals tab when data is available.
The learning goals are only loaded in the course detail view, not in the whole overview, to improve performance.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with Learning Goals/Prerequisites

1. Log in to Artemis
2. Navigate to Course Overview
3. Ensure that the learning goals tab is displayed
4. Navigate to Course Administration
5. Delete all learning goals/prerequisites of the course
6. Navigate back to Course Overview
7. Ensure the learning goals tab not shown

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-overview.component.ts | 83% | 93% |
